### ISSUE TITLE
SA1402 doesn't recognize records

### ISSUE THREAD

**Issue by Arthri** [DotNetAnalyzers/StyleCopAnalyzers#3852]
The following does not trigger SA1402 "file may only contain a single type", when it should
```cs
namespace Z
{
    internal record class X
    {
    }

    internal class Y
    {
    }
}
```

**Comment by bjornhellander** [comment]
I can not reproduct this, using the latest version (1.2.0-beta.556).
What version are you using?

**Comment by bjornhellander** [comment]
That one is from 2019 and has obvious problems with new language features. The recommendation is to use the latest beta version if you want to use modern c#.

**Comment by bjornhellander** [comment]
Duplicate of #3542
